plugin/machine-learning-client.lua

```diff
@@ -0,0 +1,109 @@
+-- This is a driver code to point to different ML servers and models
+-- and seek ml_inbound_status using such models
+-- this ml driver is invoked by machine-learning-plugin-after.conf
+-- currently most of this code is from https://github.com/coreruleset/coreruleset/pull/2067/files
+
+-- Variable Declarations:
+-- setting the machine learning server URL
+local ml_server_url = 'http://127.0.0.1:5000/'
+local ml_server_url = m.getvar("TX.ml_server_url")
+-- initialising the variable to return the machine learning pass or block status
+local inbound_ml_result = 0
+
+-- Importing libraries
+local ltn12 = require("ltn12")
+local http = require("socket.http")
```

Comment:
Any reason why are these outside of `main()` function? It means it's defined globally, no matter of `local` ketyword.